Automattic/wp-calypso
# A4A: Implement Marketplace checkout page

This pull request introduces the Marketplace checkout page. The layout closely resembles the original review licenses page from Jetpack Manage, but instead of a modal, it's on a dedicated page.

<img width="1727" alt="Screenshot 2024-03-18 at 8 48 19 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/eb34594f-e6c1-4a68-9cda-977d88d0ab14">


Closes https://github.com/Automattic/jetpack-genesis/issues/286

## Proposed Changes

* Add Checkout page controller context and routing.
* Link the Checkout button in the Shopping cart to the new page.
* Implement the checkout page based on the design specs. Some of the components from the previous Review licenses page have been reused.

## Testing Instructions
* Switch branch: `git checkout add/a4a/marketplace-checkout-page`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Use the live link and go to/marketplace
* Select some items and click the shopping cart toggle to reveal the Checkout button. Press it.
* Confirm that the page is redirected to the Checkout page. The checkout page should display items listed in the shopping cart with the correct price breakdown.
* Confirm that the **Empty cart** and **Continue shopping** buttons are working as expected.

***Note that the Purchase button does not work and will require you to set up an A4A Agency account. This process will be simplified once we have merged the PR that integrates the Agency API to A4A Authentication.***

## Pre-merge Checklist
